### PR TITLE
[WIP] vim-patch:7.4.1981

### PR DESCRIPTION
**Pull request by lonerover**
#### vim-patch:7.4.1981

Problem:    No testing for Farsi code.
Solution:   Add a minimal test.  Clean up Farsi code.

https://github.com/vim/vim/commit/ee2615af64fdcee87d8e4b13b65356e77fbd969b

### CODE REVIEW

**Comment by lonerover** [comment]
The farsi test requires 'set enc=latin', but as "utf-8" is always used now, the test failed.
How should we deal with this test?

File: src/nvim/testdir/test_farsi.vim [hunk 1]
```diff
@@ -0,0 +1,84 @@
+" Simplistic testing of Farsi mode.
+
+if !has('farsi')
+  finish
+endif
+" Farsi uses a single byte encoding.
+set enc=latin1
```
Comment (justinmk) [inline src/nvim/testdir/test_farsi.vim]: if you remove this does it fail?

Comment (lonerover) [inline src/nvim/testdir/test_farsi.vim]: yes, it failed.

File: src/nvim/testdir/test_farsi.vim [hunk 2]
```diff
@@ -0,0 +1,84 @@
+" Simplistic testing of Farsi mode.
+
+if !has('farsi')
```
Comment (justinmk) [inline src/nvim/testdir/test_farsi.vim]: let's change this to:

```vim
if !has('farsi') || has('nvim')  " Not supported in Nvim. #6192
```

**Comment by justinmk** [comment]
If farsi mode doesn't work with UTF-8 then we can't support it. Reading the "Restrictions" section at `:help farsi`, it seems even Vim doesn't care too much about this mode. I wonder if anyone uses it (in Vim). It may be another thing Vim is proud of but is actually just a trophy.

For this PR, let's disable the test.

**Comment by justinmk** [comment]
Thanks @lonerover

**Pull request merged**